unit/test_tags.py

```diff
@@ -0,0 +1,123 @@
+            'Expel instance',
+            'Set one control instance to rule them all',
+            'Set control instance facts',
+            'Cartridge auth',
+            'Application config',
+            'Bootstrap vshard',
+            'Manage failover',
+        ])
+
+    def test_tag_cartridge_instances(self):
+        names = self.get_tasks_by_tags({'cartridge-instances'})
+        self.assertEqual(names, [
+            'Check distribution',
+            'Set remote_user for delegated tasks',
+            'Validate config',
+            'Set instance facts',
+            'Install package',
+            'Start instance',
+            'Restart instances and reload systemd-daemon',
+            'Wait for instance to be started',
+            'Connect instance to membership',
+        ])
+
+    def test_tag_cartridge_replicasets(self):
+        names = self.get_tasks_by_tags({'cartridge-replicasets'})
+        self.assertEqual(names, [
+            'Check distribution',
+            'Set remote_user for delegated tasks',
+            'Validate config',
+            'Set instance facts',
+            # TODO: Remove 'Restart instances and reload systemd-daemon' when Ansible will be updated to 2.11 or higher
+            # Ref: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/meta_module.html#notes
+            'Restart instances and reload systemd-daemon',
+            'Connect instance to membership',
+            'Select one not expelled instance',
+            'Set one control instance to join replicasets',
+            'Get replicasets',
+            'Manage replicasets',
+            'Expel instance',
+        ])
+
+    def test_tag_cartridge_config(self):
+        names = self.get_tasks_by_tags({'cartridge-config'})
+        self.assertEqual(names, [
+            'Check distribution',
+            'Set remote_user for delegated tasks',
+            'Validate config',
+            'Set instance facts',
+            # TODO: Remove 'Restart instances and reload systemd-daemon' when Ansible will be updated to 2.11 or higher
+            # Ref: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/meta_module.html#notes
+            'Restart instances and reload systemd-daemon',
+            'Select one not expelled instance',
+            'Set one control instance to rule them all',
+            'Set control instance facts',
+            'Cartridge auth',
+            'Application config',
+            'Bootstrap vshard',
+            'Manage failover',
+        ])
+
+    def test_not_cartridge_tag(self):
+        names = self.get_tasks_by_tags({'test'})
+        self.assertEqual(names, [
+            # TODO: Remove 'Restart instances and reload systemd-daemon' when Ansible will be updated to 2.11 or higher
+            # Ref: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/meta_module.html#notes
+            'Restart instances and reload systemd-daemon',
+        ])
```

Comment:
It's very-very good that now tags are covered by tests, thanks!

But I worry about copy-paste of tasks names - each change of task name will lead to many changes. I suggest grouping tasks names in variables. Smth like 

```python
COMMON_TASKS = [
            'Check distribution',
            'Set remote_user for delegated tasks',
            'Validate config',
            'Set instance facts',
]

INSTANCE_TASKS = [
            'Install package',
            'Start instance',
            'Restart instances and reload systemd-daemon',
            'Wait for instance to be started',
]

REPLICASETS_TASKS = ...
```

and in tests

```python
self.assertEqual(names, COMMON_TASKS + INSTANCE_TASKS)
```

I think that it would be more clear than tasks lists.

Reply:
However, some tasks have multiple tags, but not all. For example, `Select one not expelled instance` has the tags` cartridge-replicasets` and `cartridge-config` and does not have the tag` cartridge-instances`. If you want to group them, we should use a dictionary like this:
```
{
   "Select one not expelled instance": ["cartridge-replicasets", "cartridge-config"],
}
```
And then make lists for each tag from this dictionary.
But this variant is not as readable as the current one. We will also duplicate the Ansible logic, which is not a good idea.

Reply:
So this is a good test for checking the logic of tags without calculation, which is more correct for tests.